### ISSUE TITLE
 v2.0.4-6-g111eab0 still getting GetPlayerInfoByGUID errors

### ISSUE THREAD

**Issue by ceylina** [nebularg/Parrot2#17]
Was in a neths lair in mythic plus and this occurred twice

2x Parrot\Data\CombatEvents.lua:124: Usage: GetPlayerInfoByGUID("playerGUID")
[C]: in function `GetPlayerInfoByGUID'
Parrot\Data\CombatEvents.lua:124: in function `value'
Parrot\Code\CombatEvents.lua:1658: in function <Parrot\Code\CombatEvents.lua:1653>
[C]: in function `gsub'
Parrot\Code\CombatEvents.lua:2058: in function <Parrot\Code\CombatEvents.lua:2020>
Parrot\Code\CombatEvents.lua:2178: in function <Parrot\Code\CombatEvents.lua:2175>

Locals:

**Comment by ceylina** [comment]
Do you think something like this in the function retrieveSourceName(info) would help?

if info.sourceID == nil then return end

Since you can sometimes get a nil result for the GUID? Like what if in some of these dungeons it gets confused and assumes the GUID is a non player "unit" and returns nil thus freaking out the GetPlayerInfoByGUID function?

**Comment by nebularg** [comment]
[`not info.sourceID`](https://github.com/nebularg/Parrot2/blob/bbbe65c26201c4cbecd06df49049a74469605c28/Data/CombatEvents.lua#L118) is basically the same as `info.sourceID == nil` so I doubt it's that. I have a bunch of debug around it, but as you've probably guessed I haven't been playing much in the past few months. I had originally added some checking to the guid (making sure the first part is "Player'), but I removed that and didn't include a `UnitIsPlayer` check. I'll try to run some instances in the next few days, but try adding 
```lua
	if not UnitIsPlayer(info.sourceName) then
		return info.sourceName
	end
```
before GetPlayerInfoByGUID and see if that prevents the error

**Comment by ceylina** [comment]
The odd thing is it seems to only happen in very random and odd circumstances in dungeons. I'll give that a try and see if it prevents this issue.

**Comment by ceylina** [comment]
Ok so after running everything I can think of to get this to break again, I have not had a single issue since adding your fix. I think that should take care of this as well as it can.

**Comment by ceylina** [comment]
I am pretty sure you rolled this into the latest update so is it safe to close this as I have not had a single issue with the latest update (2.0.6 i think it is?)

**Comment by nebularg** [comment]
oops, commit closed #7 instead of #17 :unamused:
